examples/source_separation/conv_tasnet/train.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python3
+"""Train Conv-TasNet"""
+import time
+import pathlib
+import argparse
+
+import torch
+import torchaudio
+
+import conv_tasnet
+from utils import dataset_utils, dist_utils
+
+_LG = dist_utils.getLogger(__name__)
+
+
+def _parse_args(args):
```

Comment:
Split the CLI interface into a separate file. Treat the training loop as a library (engine or whatever). That way you're separating the user interface from the implementaiton.